pkg/controllers/options/tagging_controller.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package options
+
+import (
+	"fmt"
+	"github.com/spf13/pflag"
+)
+
+type TaggingControllerOptions struct {
+	Tags      map[string]string
+	Resources []string
+}
+
+func (o *TaggingControllerOptions) AddFlags(fs *pflag.FlagSet) {
+	fs.StringToStringVar(&o.Tags, "tags", o.Tags, "Tags to apply to AWS resources in the tagging controller.")
+	fs.StringArrayVar(&o.Resources, "resources", o.Resources, "AWS resources name to add/remove tags in the tagging controller.")
+}
+
+func (o *TaggingControllerOptions) Validate() error {
+	if len(o.Tags) == 0 {
+		return fmt.Errorf("--tags must not be empty and must be a form of key=value")
+	}
+
+	if len(o.Resources) == 0 {
+		return fmt.Errorf("--resources must not be empty")
+	}
+
+	for _, r := range o.Resources {
+		found := false
+
+		for _, resource := range SupportedResources {
+			if r == resource {
+				found = true
```

Comment:
nit: if you're not storing each value that doesn't match the supported resources, then just return the error from inside the if.

Reply:
nit: Also, for a very slight readability improvement you could use a map or set for supportedResources so that the check is a lookup instead of an inner loop.

Reply:
got it, will update.
Regarding a map, I was using it in one of the previous commit but was told this is easier to read. (https://github.com/kubernetes/cloud-provider-aws/pull/308/commits/2f2f639102db36602f0875d19de05888c92fb103)

Reply:
Ok, I guess your reviewers have a difference of opinion then :) I'm thinking the checks if it is in the map don't have to loop over it, but its not important.  Up to you.

Reply:
A set would have perfectly solved this but with map, we had to earlier loop through it to print out a valid list of inputs which wasn't very readable in my opinion (and just printing the map as is would not have been pretty). Not something I am super concerned about too. :)